Add draft for service-vs-library pattern.

MaineC opened the pull request:
As requested in #89 this is a first draft of the service-vs-library pattern that I observed in DevOps teams.

Feedback, suggestions etc. welcome.

MaineC committed:
Add draft for service-vs-library pattern.

Fixes #89

NewMexicoKid commented on service-vs-library.md:
```diff
@@ -0,0 +1,82 @@
+support. This poses a challenge when working across team boundaries: Escalation
+chains may not be the same for errors happening in either team. Coupling
+source code and deployment leaves the teams with the question of who is
+responsible for on-call support in the event of errros. As a result teams are
```
typo: errros -> errors

rrrutledge commented on service-vs-library.md:
```diff
@@ -0,0 +1,82 @@
+Decouple configuration and deployment pipelines from actual business logic.
+Establish a second deployment of the service for the second team.
+
+Treat the common base as a library that is used by both teams with shared code
```
The **Patlet** talks about `deploying the same service in independent environments` yet the solution talks about factoring out shared service code into a library, which becomes the focus of collaboration (not the service).  I feel like these two sections of the document are incongruent with each other.  Are the teams collaborating on the source code for a library or a service?

MaineC committed:
Fix patlet to better match the solutions section.

rrrutledge committed:
Update service-vs-library.md

`erros` => `errors`

rrrutledge commented on service-vs-library.md:
```diff
@@ -0,0 +1,83 @@
+chains may not be the same for errors happening in either team. Coupling
+source code and deployment leaves the teams with the question of who is
+responsible for on-call support in the event of errors. As a result teams are
+reluctant to join forces even if there is significant overlap in requirements.
```
Another way of getting around this problem is [30-day warranty](https://github.com/paypal/InnerSourcePatterns/blob/master/30-day-warranty.md).  Could be good to mention that somewhere in this document.

rrrutledge replied:
Thanks.

rrrutledge committed:
Update service-vs-library.md

"interdependent" => "common".  I think that "common" is a better word here since the idea is to have a single code base, rather than "interdependent" which signifies multiple code basis that are mutually dependent one on another.

rrrutledge reviewed (commented):
Seems pretty clear to me.  Just one comment.

Also, I accidentally made a commit (`cbb2bb2`) directly to this branch that I meant to make as a pull request suggestion to the branch.  If you don't like it then feel free to revert.

MaineC committed:
Add link to 30 day warranty pattern

rrrutledge reviewed (approved):
Looks good to me.

NewMexicoKid reviewed (approved):
I think this is solid enough to move into the accepted patterns in the main branch. We can pursue a second review at need if people reading the pattern open a PR with suggested changes.
